pkgs/development/libraries/libgudev/default.nix

```diff
@@ -46,6 +45,10 @@ stdenv.mkDerivation (finalAttrs: {
     umockdev
   ];
 
+  mesonFlags = [
+    (lib.mesonEnable "tests" finalAttrs.finalPackage.doCheck)
```

Comment:
Why not just `finalAttrs.doCheck`?

Reply:
Idk, `finalAttrs.doCheck` was still `true` and so didn't work, and I didn't look into it further 🤷🏻‍♀️

Reply:
Oh, it's because until it goes through `mkDerivation`, the `doCheck` attr still hasn't been transformed. It looks like `finalAttrs` is resolved after all overrides, but `finalPackage` is after the `mkDerivationSimple` transformations